hive/operation.go

```diff
@@ -118,3 +118,21 @@ func (op *Operation) Close(ctx context.Context) error {
 	op.hive.log.Printf("close operation: %v", guid(op.h.OperationId.GUID))
 	return nil
 }
+
+// Cancel cancels operation
+func (op *Operation) Cancel(ctx context.Context) error {
```

Comment:
we added a new cancel API here. Does this follow the go spec? https://pkg.go.dev/database/sql
is the method signature of this method defined in the go spec we follow? could you please share the link to the cancel signature from the go spec?

Reply:
As discussed offline, `func Cancel()` would introduce a new Cancel API that does not follow the go spec in database/sql. Instead, will make this a private implementation method and already have work items for Async Execution and Context Cancellation support that will utilize this private method.

Reply:
Thanks @mattdeekay please create a work item internally to track what we discussed for following database/sql spec for cancel.